medusa/process_tv.py

```diff
@@ -432,14 +430,20 @@ def unrar(self, path, rar_files, force=False):
                 self._log('Unpacking archive: {0}'.format(archive), logger.DEBUG)
 
                 failure = None
+                # Use custom unrar path only if set, else will use from $PATH
+                if app.UNRAR_TOOL:
+                    rarfile.UNRAR_TOOL = app.UNRAR_TOOL
                 try:
-                    rar_handle = RarFile(os.path.join(path, archive))
+                    rar_handle = rarfile.RarFile(os.path.join(path, archive))
+
+                    # Test rar, if fails it will raise an exception
+                    rar_handle.testrar()
 
                     # Skip extraction if any file in archive has previously been extracted
                     skip_extraction = False
                     for file_in_archive in [os.path.basename(each.filename)
                                             for each in rar_handle.infolist()
```

Comment:
Because when the API provides a method to do **exactly** what you need, you shouldn't be writing custom code. That is the very basic thought behind using an API. Even people that never coded in their lives would understand that.